matrix/common/aws/dynamo_handler.py

```diff
@@ -148,18 +153,23 @@ def write_request_error(self, request_hash: str, message: str):
             ExpressionAttributeValues={':m': message}
         )
 
-    def get_table_item(self, table: DynamoTable, request_hash: str):
+    def get_table_item(self, table: DynamoTable, request_hash: str="", request_id: str=""):
```

Comment:
This looks a little messy to be combining request_hash and request_id params alongside the check in line 165. Why not have two separate functions that makes the logic clearer?

Reply:
I agree having two separate primary keys is messier, but I chose to build on top of the same function since we're already selecting the `DynamoTable`. I think having `get_table_item` and `get_cache_table_item` would also be messy. Alternatively, I also considered making the choice between `request_hash` and `request_id` implicit (determined by `DynamoTable`) and the parameter could be called `primary_key`. Thoughts/preferences?

Reply:
This makes sense. I'll leave this up to your judgement

Reply:
Added error handling to current implementation to ensure correct usage